feat: add customAttribute option to configure inline trigger attribute

hamedbaatour opened the pull request:
- added `customAttribute` option to allow users to set a custom attribute for inlining source files
- default attribute of course is kept `inline-source`
- updated README

bienzaaron commented:
can you `npm run test` and then check in the resulting edits to https://github.com/bienzaaron/vite-plugin-inline-source/blob/main/src/__tests__/__snapshots__/index.spec.ts.snap? otherwise, looks good to me :+1:

hamedbaatour commented:
oops, I accidentally pushed some changes earlier that weren’t meant to go in yet.
I cleaned that up and pushed again. 
and while I was at it, I added the missing tests, so we’re now at 100% coverage including checks to verify the plugin works in dev mode (not just during build) and tests to cover error handling.

<img width="374" height="94" alt="image" src="https://github.com/user-attachments/assets/08231280-532c-4e57-90f9-2bd5a9b66284" />

normally I’d split those into a separate PR, but since they’re small  (verify that plugin works in dev mode not &  add tests to cover error handling) I kept everything together.

also, please ignore the crypto polyfill I added earlier (I dropped it) I was debugging a Node issue that turned out not to be needed.

this time I ran the full pipeline script locally before pushing, so everything should be green and ready to publish.

<img width="770" height="78" alt="image" src="https://github.com/user-attachments/assets/de5a5a6b-e125-4834-be82-225bae680f92" />

<img width="441" height="78" alt="image" src="https://github.com/user-attachments/assets/d87172cf-934c-4d02-8ea0-80bf455bfcca" />

please let me know if I missed something or if anything needs fixing

bienzaaron commented:
Looks good, thanks!
